internal/clients/http/ec2/ec2_client.go

```diff
@@ -172,6 +173,33 @@ func (c *ec2Client) ImportPubkey(ctx context.Context, key *models.Pubkey, tag st
 	return *output.KeyPairId, nil
 }
 
+func (c *ec2Client) GetPubkeyName(ctx context.Context, fingerprint string) (string, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "fetchPubkeyName")
+	defer span.End()
+
+	if !c.assumed {
+		return "", http.ServiceAccountUnsupportedOperationErr
+	}
+	logger := logger(ctx)
+	logger.Trace().Msgf("Fetching AWS key with fingerprint '%s' to get its name", fingerprint)
+	input := &ec2.DescribeKeyPairsInput{}
+	input.Filters = []types.Filter{{Name: ptr.To("fingerprint"), Values: []string{fingerprint}}}
+	output, err := c.ec2.DescribeKeyPairs(ctx, input)
+	if err != nil {
+		if isAWSUnauthorizedError(err) {
+			err = clients.UnauthorizedErr
+		}
+		span.SetStatus(codes.Error, err.Error())
+		return "", fmt.Errorf("cannot fetch SSH key to update its tag %s: %w", fingerprint, err)
+	}
+
+	if len(output.KeyPairs) == 0 {
+		span.SetStatus(codes.Error, fmt.Sprintf("no KeyPair with fingerpring (%s) found", fingerprint))
+		return "", fmt.Errorf("SSH key not found by its fingerprint (%s): %w", fingerprint, http.PubkeyNotFoundErr)
+	}
+	return *output.KeyPairs[0].KeyName, nil
+}
```

Comment:
I am not sure if we should treat the 404 like a hard error, at most this should be warning, if ever.

Other than that, looks good.

Reply:
well the error gets caught in the only caller, but if you think we should just return empty string, I'm ok with it, but feels bit weird, so I thought error is bit cleaner there.

Reply:
Happy to indicate not found in other means, but would you have a suggestion?

Reply:
Nah you are right that the caller should be responsible for skipping errors. This is fine.